### ISSUE TITLE
Not actually cleanly compatible with PHP 8.1

### ISSUE THREAD

**Issue by TimWolla** [MyIntervals/emogrifier#1117]
Apparently PHP deprecation errors are disabled by default when using `shivammathur/setup-php@v2`. Emogrifier is not actually cleanly compatible with PHP 8.1, due to Sabberworm CSS Parser not being compatible: https://github.com/sabberworm/PHP-CSS-Parser/pull/344

Consider setting `error_reporting=E_ALL` in GitHub Actions:

```diff
From dc7e00f23094eec7224e49f47326fab78af06931 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20D=C3=BCsterhus?= <duesterhus@woltlab.com>
Date: Thu, 28 Oct 2021 15:36:26 +0200
Subject: [PATCH] Update ci.yml

---
 .github/workflows/ci.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index ecd42ab4..1da8494c 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -157,7 +157,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php-version }}
-          ini-values: xdebug.max_nesting_level=500
+          ini-values: xdebug.max_nesting_level=500, error_reporting=E_ALL
           tools: composer:v2
           coverage: none
```

Example run: https://github.com/TimWolla/emogrifier/runs/4035243076?check_suite_focus=true

**Comment by JakeQZ** [comment]
This specific issue I hope now fixed since [CSS Parser 8.4](https://packagist.org/packages/sabberworm/php-css-parser#8.4.0) release.

Let us know either way.

**Comment by TimWolla** [comment]
> This specific issue I hope now fixed since [CSS Parser 8.4](https://packagist.org/packages/sabberworm/php-css-parser#8.4.0) release.

Looks good to me. Leaving this issue open in case you also want to up your GH actions workflow to report all types of error.
